src/main/scala/com/microsoft/hyperspace/index/sources/FileBasedSourceProviderManager.scala

```diff
@@ -90,6 +90,19 @@ class FileBasedSourceProviderManager(spark: SparkSession) {
     run(p => p.getRelation(plan))
   }
 
+  /**
+   * Returns enriched index properties.
+   *
+   * @param relation Relation to retrieve necessary information.
+   * @param previousProperties Index properties of previous index version.
```

Comment:
`previousProperties` could be misleading if you are creating an index? just `properties` seems fine to me.